plugins/inputs/nsdp/nsdp_test.go

```diff
@@ -0,0 +1,69 @@
+package nsdp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-nsdp"
+)
+
+func TestConfig(t *testing.T) {
+	// Verify plugin can be loaded from config
+	conf := config.NewConfig()
+	require.NoError(t, conf.LoadConfig("testdata/conf/nsdp.conf"))
+	require.Len(t, conf.Inputs, 1)
+	plugin, ok := conf.Inputs[0].Input.(*NSDP)
+	require.True(t, ok)
+
+	// Verify successful Init
+	require.NoError(t, plugin.Init())
+
+	// Verify everything is setup according to config file
+	require.Equal(t, "127.0.0.1:63322", plugin.Address)
+	require.Equal(t, uint(1), plugin.DeviceLimit)
+	require.Equal(t, config.Duration(5*time.Second), plugin.Timeout)
+}
```

Comment:
Hmmm this tests only the configuration parsing code, doesn't it? If so, you don't need it as this is tested a lot in the `config` package...

Reply:
This test ensures plugin typedef matches the config file and every config file parameter is read properly.